app/models/post.rb

```diff
@@ -1,13 +1,33 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
-  has_many :comments
-  has_many :likes
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
 
-  after_save :update_author_posts_counter
+  validates :title, presence: true, length: { maximum: 250 }
+  validates :commentscounter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :likescounter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+
+  after_save :increment_author_posts_counter
+  after_destroy :decrement_author_posts_counter
+
+  after_create :update_likes_counter
+  after_destroy :update_likes_counter
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
 
   private
 
-  def update_author_posts_counter
-    author.update(posts_counter: author.posts.count)
+  def increment_author_posts_counter
+    author.increment!(:posts_counter)
+  end
+
+  def decrement_author_posts_counter
+    author.decrement!(:posts_counter)
+  end
+
+  def update_likes_counter
+    update(likescounter: likes.count)
   end
```

Comment:
- [ ] You did an excellent job of testing the other methods and validations 👏. However, in order to meet the requirement: **Add unit specs for all of your models' methods and validations.** I would also recommend that you test the `update_likes_counter` method too. I believe that you forgot about it. ☺️
By testing various components and functionalities in a systematic manner, you can identify issues that may result in crashes, incorrect results, security vulnerabilities, or a poor user experience.